tools/extra/pac/fpgabist/dma/fpga_dma_test.c

```diff
@@ -54,6 +61,132 @@ static int err_cnt;
       goto label;\
     } \
   } while (0)
+/*
+ * Global configuration of bus, set during parse_args()
+ * */
+struct config{
+  struct target {
+	  int bus;
+  } target;
+} config = {
+	.target = {
+		.bus = -1
+	}
+};
+
+void help(void)
+{
+	printf(
+"\n"
+"fpga_dma_test \n"
+"Usage:\n"
+"\n"
+"               0          Hardware\n"
+"               1          Simulation only\n"
+"              -B, --bus   Set target bus number (OPTIONAL)\n"
+"\n"
+                   );
+}
+
+
+/*
+ * Parse command line arguments
+ */
+#define GETOPT_STRING "B"
+int parse_args(int argc, char *argv[])
+{
+	struct option longopts[] = {
+		
+		{"bus",          required_argument, NULL, 'B'}
+	};
+	
+	int getopt_ret;
+	int option_index;
+	char *endptr = NULL;
+
+	while (-1 != (getopt_ret = getopt_long(argc, argv, GETOPT_STRING, 
+						longopts, &option_index))) {
+		const char *tmp_optarg = optarg;
+		
+		if ((optarg) && ('=' == *tmp_optarg)){
+			++tmp_optarg;
+		}
+		
+		switch (getopt_ret) {
+		case 'B':   /* bus */
+			if (NULL == tmp_optarg)
+				break;
+			endptr = NULL;
+			config.target.bus = (int) strtoul(tmp_optarg, &endptr, 0);
+			if (endptr != tmp_optarg + strlen(tmp_optarg)) {
+				fprintf(stderr, "invalid bus: %s\n", tmp_optarg);
+				return -1;
+			}
+			break;
+	 	
+     		default: /* invalid option */
+			fprintf(stderr, "Invalid cmdline options\n");			return -1;
+		}
+	}
+
+	/* first non-option argument as hardware or simulation*/
+	if (optind == argc) {
+		fprintf(stderr, "Hardware (0) or simulation (1)?\n");
+		return -1;
+	}
+	
+	return 0;
+}
+
+
+     
+
+
+
+int find_fpga(fpga_guid interface_id, fpga_token *fpga)
```

Comment:
This function appears to have ambiguous return values. Do you want it to return error code or number of matches? Cant be both.

By convention, all functions should return error code or error count. Use an argument by reference to populate the number of matches.